packages/flutter/test/rendering/layers_test.dart

```diff
@@ -44,7 +44,7 @@ void main() {
     expect(boundary.layer.attached, isTrue); // this time it did again!
   });
 
-  test('layer subtree dirtiness is correctly computed', () {
+  test('layer _subtreeNeedsAddToScene is correctly computed', () {
```

Comment:
One more thought: if we still computes `subtreeNeedsAddToScene`, then why do we have to traverse upward during `markNeedsAddToScene`? It seems that we can still just mark the current node.

Reply:
I remove _subtreeNeedsAddToScene in favor or reusing _needsAddToScene and added tests. PTAL.